### PR TITLE
Icon generator

### LINKED ISSUE #2: should we generate icons files via script?
**TypeScript Version:** 2.0.0

**Typeings Version:** master  

**Issue:**
material-ui has a lot of svg icons files; is there a way to declare them all at once, or should we consider using a script to generate them in a batch process?

### PR DESCRIPTION
**Pull request by alitaheri**
I added the icon type generator and ran it once on Windows. I don't know how it will turn out in linux. and I haven't tested the output, this is only for review, don't merge yet (or do if it works for you :laughing: )

Closes #2

### CODE REVIEW

File: icon-typings-builder/build.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
+#! /usr/bin/env node
+/**
+ * Based on https://github.com/callemall/material-ui/tree/master/packages/icon-builder
+ */
+
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const rimraf = require('rimraf');
```
Comment (mattiamanzati) [inline icon-typings-builder/build.js]: rimraf is missing from package.json devDependencies ;)

Comment (alitaheri) [inline icon-typings-builder/build.js]: ops :sweat_smile:

File: icon-typings-builder/build.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+const glob = require('glob');
+const mkdirp = require('mkdirp');
+const rrs = require('recursive-readdir-sync');
+
+const SVG_ICON_RELATIVE_REQUIRE = '../../SvgIcon';
+const OUTPUT_DIR = path.resolve(__dirname, '../svg-icons');
+const SVG_DIR = path.resolve(__dirname, './node_modules/material-design-icons/');
+
+function main() {
+  rimraf.sync(OUTPUT_DIR); // Clean old files
```
Comment (mattiamanzati) [inline icon-typings-builder/build.js]: Maybe we should simply remove this line and configure a "clean" task in package.json

Comment (alitaheri) [inline icon-typings-builder/build.js]: Yeah i guess that's a better way to do it :+1:

**Comment by mattiamanzati** [comment]
Just ran on Ubuntu and it is perfect!
Just setup a clean task and we are ready to rule!

**Comment by alitaheri** [comment]
It's done :grin:

**Pull request merged**